app/components/attachment_button/index.test.js

```diff
@@ -76,19 +70,18 @@ describe('AttachmentButton', () => {
         });
     });
 
-    test('should show permission denied alert if permission is denied in iOS', async () => {
-        expect.assertions(1);
+    // test('should show permission denied alert if permission is denied in iOS', async () => {
```

Comment:
since the logic changed, this test can be written as two tests as follows and should pass,

    test('should return permission false if permission is denied in iOS', async () => {
        jest.spyOn(Permissions, 'check').mockReturnValue(Permissions.RESULTS.DENIED);
        jest.spyOn(Permissions, 'request').mockReturnValue(Permissions.RESULTS.DENIED);

        const wrapper = shallow(
            <AttachmentButton {...baseProps}/>,
            {context: {intl: {formatMessage}}},
        );

        const hasPhotoPermission = await wrapper.instance().hasPhotoPermission('camera');
        expect(Permissions.check).toHaveBeenCalled();
        expect(Permissions.request).toHaveBeenCalled();
        expect(Alert.alert).not.toHaveBeenCalled();
        expect(hasPhotoPermission).toBe(false);
    });

    test('should show permission denied alert and return permission false if permission is blocked in iOS', async () => {
        jest.spyOn(Permissions, 'check').mockReturnValue(Permissions.RESULTS.BLOCKED);
        jest.spyOn(Alert, 'alert').mockReturnValue(true);

        const wrapper = shallow(
            <AttachmentButton {...baseProps}/>,
            {context: {intl: {formatMessage}}},
        );

        const hasPhotoPermission = await wrapper.instance().hasPhotoPermission('camera');
        expect(Permissions.check).toHaveBeenCalled();
        expect(Permissions.request).not.toHaveBeenCalled();
        expect(Alert.alert).toHaveBeenCalled();
        expect(hasPhotoPermission).toBe(false);
    });

Reply:
it worked!!